### DISCUSSION TITLE
Use one role for multiple services

### DISCUSSION THREAD

**Discussion by xinye2005** [aws/copilot-cli#5735] [General]
We have over 20 load balanced services in 2 applications and 2 environments respectively. When we used AWS copilot to create the services, these IAM roles (EnvControllerRole, AutoScalingRole, TaskRole, RulePriorityFunctionRole & DynamicDesiredCountFunct) were created for each of the services with a unique role name. That amounts to over 400 roles. When I looked into the details of each role, say TaskRole, they're the same for each service except the tag. AWS has a quota for IAM roles. If you have hundreds of services, it'll run out of quota. Is there a way to use the same role for all the services in the same app and the same environment? Then we only need about 20 (5 x 2 x 2) over roles for our case.

**Comment by KollaAdithya** [comment]
Hello @xinye2005 !

I agree completely on this. If you want to use the same roles across all services you can use [yamlpatch overrides](https://aws.github.io/copilot-cli/docs/developing/overrides/yamlpatch/) to replace the existing roles.
